### PR TITLE
fix: Log attempts to output resp.Body, part of #104184

### LINKED ISSUE #104184: Log attempts to output resp.Body
#### What happened:

Log messages like `body of failing http response: &{0x531900 0xc0010d2780 0x775180}` are not very informative.

/sig cloud-provider

#### What you expected to happen:

Either actual response body or none of it if this is a security concern.

/sig security

sig security to confirm it is ok to change this to log an actual response body.

#### How to reproduce it (as minimally and precisely as possible):

two places it is being logged:

- https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/legacy-cloud-providers/gce/gcpcredential/credentialutil.go#L63
- https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/pkg/credentialprovider/config.go#L206

#### Environment:
- Kubernetes version (use `kubectl version`):1.20, but doesn't matter
- Cloud provider or hardware configuration:GKE
- OS (e.g: `cat /etc/os-release`):COS

### PR DESCRIPTION
**Pull request by siwet**
#### Which issue(s) this PR fixes:

Fixes #104184

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by skywe-tang** [comment]
@k8s-ci-robot verify CLA

File: pkg/credentialprovider/config.go [hunk 1]
```diff
@@ -202,20 +202,20 @@ func ReadURL(url string, client *http.Client, header *http.Header) (body []byte,
...
 
+	limitedReader := &io.LimitedReader{R: resp.Body, N: maxReadLength}
+	contents, err := ioutil.ReadAll(limitedReader)
+	if err != nil {
+		return nil, err
+	}
+
 	if resp.StatusCode != http.StatusOK {
-		klog.V(2).Infof("body of failing http response: %v", resp.Body)
+		klog.V(2).Infof("body of failing http response: %s", contents)
```
Comment (kerthcet) [inline pkg/credentialprovider/config.go:212]: I was doubt about the security of this code segment. As `ReadDockerConfigFileFromURL` will call this function, which may expose some auth info. correct me if i'm wrong.

Comment (skywe-tang) [reply pkg/credentialprovider/config.go:212]: Hi, @kerthcet, this is a draft PR  for my first submit test.
What I consider is that if the request has failed, there should be no auth info, I don’t know if it's correct. 
waiting for sig/security to confirm #104184

Comment (yuzhiquan) [reply pkg/credentialprovider/config.go:212]: May be `InfoS` will be easy to resolve this https://github.com/kubernetes/kubernetes/issues/104184#issuecomment-893935406

**Comment by skywe-tang** [comment]
Continue at #104184

/close

**Pull request closed without merging**